docs/source/transaction_family_specifications/track_and_trade_transaction_family.rst

```diff
@@ -0,0 +1,717 @@
+       repeated AssociatedAgent owners = 3;
+       repeated AssociatedAgent custodians = 4;
+
+       // Flag indicating whether the Record can be updated. If it is set
+       // to true, then the record has been finalized and no further
+       // changes can be made to it or its Properties.
+       bool final = 5;
+   }
+
+
+Note that while information about a Record's owners and custodians are
+included in the object, information about its Properties are stored
+separately (see below).
+
+Records whose addresses collide will be stored in a list
+alphabetically by identifier.
+
+.. code-block:: protobuf
+
+   message RecordContainer {
+       repeated Record entries = 1;
+   }
+
+
+Properties
+----------
+
+Historical data pertaining to a particular data field of a tracked
+object will be stored as Properties, represented as a list of values
+accompanied by a timestamp and a reporter identifier.
+
+The whole history of updates to Record data is stored in current state
+because this allows for more flexibility in writing transaction rules.
+For example, there might be a rule that no fish can be exchanged whose
+temperature has gone above 40 degrees. This means, however, that it
+would be impractical to store all of a Record's data at one address,
+since adding a single update would require reading the entire history
+of each of the record's properties out of state, adding the update,
+then writing it all back.
+
+To solve this problem, Properties will be stored in their own
+namespace derived from their name and associated Record. Since some
+Properties may have thousands of updates, four characters will be
+reserved at the end of that namespace in order to paginate a
+Property's history. The Property itself, with name, Record identifier,
+authorized reporters, and paging information, will be stored at the
+namespace ending in `0000`. The namespaces ending in `0001` to `ffff`
+will each store a PropertyPage containing up to 256 reported values
+(which include timestamps and their reporter's identity). Any
+Transaction updating the value of a Property first reads out the
+PropertyContainer object at `0000` and then reads out the appropriate
+PropertyPageContainer before adding the update and writing the new
+PropertyPageContainer back to state.
+
+The Transaction Processor will treat these pages as a ring buffer, so
+that when page `ffff` is filled, the next update will erase the
+entries at page `0001` and be stored there, and subsequent
+page-filling will continue to overwrite the next oldest page. This
+will ensure no Property ever runs out of space for new updates. Under
+this scheme, 16^2 * (16^4 - 1) = 16776960 entries can be stored before
+older updates are overwritten.
+
+Updates to Properties can have one of the following protobuf types:
+`bytes`, `string`, `sint64`, `float`, or `Location` (see the section
+on RecordTypes below). The type of an update is indicated by a tag
+belonging to the PropertySchema object.
```

Comment:
Awkward wording. Suggest: _". . . is indicated by the DataType enum in the PropertySchema message"_.